### PR TITLE
Unignore the pulumi code and add a test for it

### PR DESCRIPTION
**Pull request by jrockway**
There was an issue with a dependency of Pulumi that prevented us from building with Bazel.  That is fixed, or at least worked around cleanly: https://github.com/bazelbuild/rules_go/issues/3799#issuecomment-1880035642

With the workarounds, this now builds OK, so I've unignored it and added a BUILD file.

There was also an issue where it generates invalid helm values.  I fixed that bug and added a test.

### CODE REVIEW

**Comment by davidfluck-hpe** [comment]
Sorry, catching up after an appointment this morning. Looking over this now.

File: etc/testing/circle/workloads/pulumi/aws/app.go [hunk 1]
```diff
@@ -190,13 +190,13 @@ func DeployApp(ctx *pulumi.Context, k8sProvider *kubernetes.Provider, saRole *ia
...
+				"backend":      pulumi.String("AMAZON"),
+				"gocdkEnabled": pulumi.Bool(true),
+				"storageURL":   pulumi.Sprintf("s3://%s", bucket.Bucket),
 				"amazon": pulumi.Map{
-					"gocdkEnabled": pulumi.Bool(true),
-					"storageURL":   pulumi.Sprintf("s3://%s", bucket.Bucket),
-					"region":       pulumi.String("us-west-2"),
-					"id":           pulumi.String(awsSAkey),
-					"secret":       pulumi.String(awsSAsecret),
+					"region": pulumi.String("us-west-2"),
```
Comment (FahadBSyed) [inline etc/testing/circle/workloads/pulumi/aws/app.go:197]: FWIW you can make the region a url query parameter on the storageURL `s3://bucket?region=us-west-2`

**Review by FahadBSyed — APPROVED** [review]
only one nit, otherwise looks good!

File: etc/testing/circle/workloads/pulumi/aws/app.go (hunk 1 above)
Comment (jrockway) [reply etc/testing/circle/workloads/pulumi/aws/app.go]: Done.

**Review by davidfluck-hpe — APPROVED** [review]
Looks good, works locally for me.

**Comment by armfazh** [comment]
The compilation issues of CIRCL are solved by updating [rules_go v0.55.0](https://github.com/bazel-contrib/rules_go/releases/tag/v0.55.0). See more details at CIRCL [Bazel-Compilation](https://github.com/cloudflare/circl/wiki/Bazel-Compilation) wiki.

**Pull request merged**